File: solutions/python/robot-name/1/robot_name.py
import string
import random
class Robot:
    robots=[]
    def __init__(self):
        self.name = self.gen_name()

    def gen_name(self):    
        unique_name = False
        while not unique_name:
            name=random.choice(string.ascii_uppercase) + \
                    random.choice(string.ascii_uppercase) + \
                    random.choice('0123456789') + random.choice('0123456789') + \
                    random.choice('0123456789')
            #print("gen_name",name)
            if name not in self.robots:
                unique_name = True
                self.robots.append(name)
        return(name)        


    def reset(self):
        self.name=self.gen_name()
        #print("gen_name=",self.gen_name())
        #print(self.name)

#r=Robot()
##print (r.name)
#r.reset()
#print(r.name)

File: solutions/csharp/robot-name/1/RobotName.cs
using System;
using System.Collections.Generic;

public class Robot
{
    private string name;
    static List<string> names = new List<string>();

    public Robot () {
        Random rnd = new Random();
        char char1 = (char)('A' + rnd.Next(26));
        char char2 = (char)('A' + rnd.Next(26));
        name = String.Format("{0}{1}{2:D3}", char1.ToString(), char2.ToString(), rnd.Next(999));
        while (names.IndexOf(name)>=0) {
            name = String.Format("{0}{1}{2:D3}", char1.ToString(), char2.ToString(), rnd.Next(999));
        }
        names.Add(name);
    }


    public string Name
    {
        get => name;

    }

    public void Reset()
    {
        Random rnd = new Random();
        char char1 = (char)('A' + rnd.Next(26));
        char char2 = (char)('A' + rnd.Next(26));
        name = String.Format("{0}{1}{2:D3}", char1.ToString(), char2.ToString(), rnd.Next(999));
        while (names.IndexOf(name)>=0) {
            name = String.Format("{0}{1}{2:D3}", char1.ToString(), char2.ToString(), rnd.Next(999));
        }
        names.Add(name);
    }

}